common/ast/expr.go

```diff
@@ -606,6 +609,15 @@ func (e *baseListExpr) Elements() []Expr {
 	return e.elements
 }
 
+func (e *baseListExpr) IsOptional(index int32) bool {
+	for _, optIndex := range e.OptionalIndices() {
```

Comment:
Would it be feasible to create a map equivalent and access that instead?

Reply:
Yes, sure. I left it as an exercise for the future if this proves to be a hotspot, but can do it now.